File: frontend/src/api/logs.js
/**
 * 日志管理相关API接口
 */
import api from './index'

/**
 * 获取日志统计信息
 * @param {Object} params - 查询参数
 * @param {string} params.period - 统计周期 (1h, 24h, 7d, 30d)
 * @param {string} params.log_type - 日志类型
 * @returns {Promise}
 */
export const getLogStats = (params = {}) => {
  return api.get('/logs/stats/', { params })
}

/**
 * 获取日志汇总统计
 * @returns {Promise}
 */
export const getLogSummary = () => {
  return api.get('/logs/stats/summary/')
}

/**
 * 获取系统日志
 * @param {Object} params - 查询参数
 * @param {number} params.page - 页码
 * @param {number} params.size - 每页大小
 * @param {string} params.level - 日志级别
 * @param {string} params.category - 日志分类
 * @param {string} params.start_time - 开始时间
 * @param {string} params.end_time - 结束时间
 * @param {string} params.search - 搜索关键词
 * @returns {Promise}
 */
export const getSystemLogs = (params = {}) => {
  return api.get('/logs/system/', { params })
}

/**
 * 获取系统日志详情
 * @param {number} logId - 日志ID
 * @returns {Promise}
 */
export const getSystemLog = (logId) => {
  return api.get(`/logs/system/${logId}/`)
}

/**
 * 获取操作日志
 * @param {Object} params - 查询参数
 * @param {number} params.page - 页码
 * @param {number} params.size - 每页大小
 * @param {string} params.action - 操作类型
 * @param {string} params.status - 操作状态
 * @param {string} params.resource_type - 资源类型
 * @param {string} params.resource_id - 资源ID
 * @param {string} params.user_id - 用户ID
 * @param {string} params.start_time - 开始时间
 * @param {string} params.end_time - 结束时间
 * @param {string} params.search - 搜索关键词
 * @returns {Promise}
 */
export const getOperationLogs = (params = {}) => {
  return api.get('/logs/operations/', { params })
}

/**
 * 获取操作日志详情
 * @param {number} logId - 日志ID
 * @returns {Promise}
 */
export const getOperationLog = (logId) => {
  return api.get(`/logs/operations/${logId}/`)
}

/**
 * 获取MCP协议日志
 * @param {Object} params - 查询参数
 * @param {number} params.page - 页码
 * @param {number} params.size - 每页大小
 * @param {number} params.tool_id - 工具ID
 * @param {string} params.direction - 消息方向
 * @param {string} params.method - 方法名
 * @param {string} params.start_time - 开始时间
 * @param {string} params.end_time - 结束时间
 * @param {string} params.search - 搜索关键词
 * @returns {Promise}
 */
export const getMcpLogs = (params = {}) => {
  return api.get('/logs/mcp/', { params })
}

/**
 * 获取MCP协议日志详情
 * @param {number} logId - 日志ID
 * @returns {Promise}
 */
export const getMcpLog = (logId) => {
  return api.get(`/logs/mcp/${logId}/`)
}

/**
 * 搜索日志
 * @param {Object} params - 搜索参数
 * @param {string} params.query - 搜索查询
 * @param {string} params.log_type - 日志类型
 * @param {number} params.page - 页码
 * @param {number} params.size - 每页大小
 * @param {string} params.start_time - 开始时间
 * @param {string} params.end_time - 结束时间
 * @returns {Promise}
 */
export const searchLogs = (params = {}) => {
  return api.get('/logs/search/', { params })
}

/**
 * 获取实时日志
 * @param {Object} params - 查询参数
 * @param {string} params.log_type - 日志类型
 * @param {string} params.level - 日志级别
 * @param {number} params.limit - 限制数量
 * @returns {Promise}
 */
export const getRealtimeLogs = (params = {}) => {
  return api.get('/logs/realtime/', { params })
}

/**
 * 清理日志
 * @param {Object} data - 清理参数
 * @param {number} data.retention_days - 保留天数
 * @param {Array} data.levels - 要清理的日志级别
 * @param {Array} data.categories - 要清理的日志分类
 * @param {Array} data.log_types - 要清理的日志类型
 * @returns {Promise}
 */
export const cleanupLogs = (data) => {
  return api.post('/logs/cleanup/', data)
}

/**
 * 导出日志
 * @param {Object} params - 导出参数
 * @param {string} params.log_type - 日志类型
 * @param {string} params.format - 导出格式 (csv, json)
 * @param {string} params.start_time - 开始时间
 * @param {string} params.end_time - 结束时间
 * @param {string} params.level - 日志级别
 * @param {string} params.category - 日志分类
 * @returns {Promise}
 */
export const exportLogs = (params = {}) => {
  return api.get('/logs/export/', { params, responseType: 'blob' })
}

/**
 * 获取所有日志（统一接口）
 * @param {Object} params - 查询参数
 * @param {number} params.page - 页码
 * @param {number} params.size - 每页大小
 * @param {string} params.logType - 日志类型 (system, operation, mcp)
 * @param {string} params.level - 日志级别
 * @param {string} params.source - 来源
 * @param {string} params.search - 搜索关键词
 * @param {Array} params.dateRange - 时间范围 [start, end]
 * @returns {Promise}
 */
export const getAllLogs = (params = {}) => {
  const { logType, dateRange, ...otherParams } = params
  
  // 处理时间范围
  if (dateRange && dateRange.length === 2) {
    otherParams.start_time = new Date(dateRange[0]).toISOString()
    otherParams.end_time = new Date(dateRange[1]).toISOString()
  }
  
  // 根据日志类型调用不同的API
  switch (logType) {
    case 'system':
      return getSystemLogs(otherParams)
    case 'operation':
      return getOperationLogs(otherParams)
    case 'mcp':
      return getMcpLogs(otherParams)
    default:
      // 默认获取系统日志
      return getSystemLogs(otherParams)
  }
}

export default {
  getLogStats,
  getLogSummary,
  getSystemLogs,
  getSystemLog,
  getOperationLogs,
  getOperationLog,
  getMcpLogs,
  getMcpLog,
  searchLogs,
  getRealtimeLogs,
  cleanupLogs,
  exportLogs,
  getAllLogs
}

File: backend/app/api/logs.py
"""日志管理 API 路由"""

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List, Optional
from datetime import datetime, timedelta
import logging

from app.core.database import get_db
from app.schemas.log import (
    SystemLogResponse,
    OperationLogResponse,
    MCPLogResponse,
    LogQueryParams,
    LogStatsResponse,
    LogCleanupParams,
    PaginatedResponse,
)
from app.services.system import LogService
from app.utils.response import success_response, error_response
from app.utils.pagination import simple_paginate as paginate

logger = logging.getLogger(__name__)
router = APIRouter(prefix="/logs", tags=["日志管理"])

# 系统日志
@router.get("/system", response_model=dict, summary="获取系统日志")
async def get_system_logs(
    page: int = Query(1, ge=1, description="页码"),
    size: int = Query(50, ge=1, le=200, description="每页大小"),
    level: Optional[str] = Query(None, description="日志级别"),
    category: Optional[str] = Query(None, description="日志分类"),
    start_time: Optional[datetime] = Query(None, description="开始时间"),
    end_time: Optional[datetime] = Query(None, description="结束时间"),
    search: Optional[str] = Query(None, description="搜索关键词"),
    db: Session = Depends(get_db)
):
    """获取系统日志"""
    try:
        log_service = LogService(db)

        # 构建过滤条件
        filters = {}
        if level:
            filters['level'] = level
        if category:
            filters['category'] = category
        if start_time:
            filters['start_time'] = start_time
        if end_time:
            filters['end_time'] = end_time
        if search:
            filters['search'] = search

        # 获取日志列表
        logs, total = log_service.get_system_logs(
            page=page,
            size=size,
            filters=filters
        )

        # 分页信息
        pagination = paginate(total, page, size)

        return success_response(
            data={
                "items": [log.to_dict() for log in logs],
                "pagination": pagination
            },
            message="获取系统日志成功"
        )
    except Exception as e:
        logger.error(f"获取系统日志失败: {e}")
        return error_response(message="获取系统日志失败", error_code=str(e))

@router.get("/system/{log_id}", response_model=dict, summary="获取系统日志详情")
async def get_system_log(
    log_id: int,
    db: Session = Depends(get_db)
):
    """获取系统日志详情"""
    try:
        log_service = LogService(db)
        log = log_service.get_system_log(log_id)

        if not log:
            raise HTTPException(status_code=404, detail="日志不存在")

        return success_response(
            data=log.to_dict(),
            message="获取系统日志详情成功"
        )
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"获取系统日志详情失败: {e}")
        return error_response(message="获取系统日志详情失败", error_code=str(e))

# 操作日志
@router.get("/operations", response_model=dict, summary="获取操作日志")
async def get_operation_logs(
    page: int = Query(1, ge=1, description="页码"),
    size: int = Query(50, ge=1, le=200, description="每页大小"),
    action: Optional[str] = Query(None, description="操作类型"),
    status: Optional[str] = Query(None, description="操作状态"),
    resource_type: Optional[str] = Query(None, description="资源类型"),
    resource_id: Optional[str] = Query(None, description="资源ID"),
    user_id: Optional[str] = Query(None, description="用户ID"),
    start_time: Optional[datetime] = Query(None, description="开始时间"),
    end_time: Optional[datetime] = Query(None, description="结束时间"),
    search: Optional[str] = Query(None, description="搜索关键词"),
    db: Session = Depends(get_db)
):
    """获取操作日志"""
    try:
        log_service = LogService(db)

        # 构建过滤条件
        filters = {}
        if action:
            filters['action'] = action
        if status:
            filters['status'] = status
        if resource_type:
            filters['resource_type'] = resource_type
        if resource_id:
            filters['resource_id'] = resource_id
        if user_id:
            filters['user_id'] = user_id
        if start_time:
            filters['start_time'] = start_time
        if end_time:
            filters['end_time'] = end_time
        if search:
            filters['search'] = search

        # 获取日志列表
        logs, total = log_service.get_operation_logs(
            page=page,
            size=size,
            filters=filters
        )

        # 分页信息
        pagination = paginate(total, page, size)

        return success_response(
            data={
                "items": [log.to_dict() for log in logs],
                "pagination": pagination
            },
            message="获取操作日志成功"
        )
    except Exception as e:
        logger.error(f"获取操作日志失败: {e}")
        return error_response(message="获取操作日志失败", error_code=str(e))

@router.get("/operations/{log_id}", response_model=dict, summary="获取操作日志详情")
async def get_operation_log(
    log_id: int,
    db: Session = Depends(get_db)
):
    """获取操作日志详情"""
    try:
        log_service = LogService(db)
        log = log_service.get_operation_log(log_id)

        if not log:
            raise HTTPException(status_code=404, detail="日志不存在")

        return success_response(
            data=log.to_dict(),
            message="获取操作日志详情成功"
        )
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"获取操作日志详情失败: {e}")
        return error_response(message="获取操作日志详情失败", error_code=str(e))

# MCP 协议日志
@router.get("/mcp", response_model=dict, summary="获取MCP协议日志")
async def get_mcp_logs(
    page: int = Query(1, ge=1, description="页码"),
    size: int = Query(50, ge=1, le=200, description="每页大小"),
    tool_id: Optional[int] = Query(None, description="工具ID"),
    direction: Optional[str] = Query(None, description="消息方向"),
    method: Optional[str] = Query(None, description="方法名"),
    start_time: Optional[datetime] = Query(None, description="开始时间"),
    end_time: Optional[datetime] = Query(None, description="结束时间"),
    search: Optional[str] = Query(None, description="搜索关键词"),
    db: Session = Depends(get_db)
):
    """获取MCP协议日志"""
    try:
        log_service = LogService(db)

        # 构建过滤条件
        filters = {}
        if tool_id:
            filters['tool_id'] = tool_id
        if direction:
            filters['direction'] = direction
        if method:
            filters['method'] = method
        if start_time:
            filters['start_time'] = start_time
        if end_time:
            filters['end_time'] = end_time
        if search:
            filters['search'] = search

        # 获取日志列表
        logs, total = log_service.get_mcp_logs(
            page=page,
            size=size,
            filters=filters
        )

        # 分页信息
        pagination = paginate(total, page, size)

        return success_response(
            data={
                "items": [log.to_dict() for log in logs],
                "pagination": pagination
            },
            message="获取MCP协议日志成功"
        )
    except Exception as e:
        logger.error(f"获取MCP协议日志失败: {e}")
        return error_response(message="获取MCP协议日志失败", error_code=str(e))

@router.get("/mcp/{log_id}", response_model=dict, summary="获取MCP协议日志详情")
async def get_mcp_log(
    log_id: int,
    db: Session = Depends(get_db)
):
    """获取MCP协议日志详情"""
    try:
        log_service = LogService(db)
        log = log_service.get_mcp_log(log_id)

        if not log:
            raise HTTPException(status_code=404, detail="日志不存在")

        return success_response(
            data=log.to_dict(),
            message="获取MCP协议日志详情成功"
        )
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"获取MCP协议日志详情失败: {e}")
        return error_response(message="获取MCP协议日志详情失败", error_code=str(e))

# 日志统计
@router.get("/stats", response_model=dict, summary="获取日志统计")
async def get_log_stats(
    period: str = Query("24h", description="统计周期"),
    log_type: Optional[str] = Query(None, description="日志类型"),
    db: Session = Depends(get_db)
):
    """获取日志统计"""
    try:
        log_service = LogService(db)

        # 解析时间周期
        if period == "1h":
            start_time = datetime.now() - timedelta(hours=1)
        elif period == "24h":
            start_time = datetime.now() - timedelta(days=1)
        elif period == "7d":
            start_time = datetime.now() - timedelta(days=7)
        elif period == "30d":
            start_time = datetime.now() - timedelta(days=30)
        else:
            start_time = datetime.now() - timedelta(days=1)

        # 获取统计数据
        stats = log_service.get_log_statistics(
            period=period
        )

        return success_response(
            data=stats,
            message="获取日志统计成功"
        )
    except Exception as e:
        logger.error(f"获取日志统计失败: {e}")
        return error_response(message="获取日志统计失败", error_code=str(e))

@router.get("/stats/summary", response_model=dict, summary="获取日志汇总统计")
async def get_log_summary(
    db: Session = Depends(get_db)
):
    """获取日志汇总统计"""
    try:
        log_service = LogService(db)
        summary = log_service.get_summary_statistics()

        return success_response(
            data=summary,
            message="获取日志汇总统计成功"
        )
    except Exception as e:
        logger.error(f"获取日志汇总统计失败: {e}")
        return error_response(message="获取日志汇总统计失败", error_code=str(e))

# 日志管理
@router.post("/cleanup", response_model=dict, summary="清理日志")
async def cleanup_logs(
    cleanup_params: LogCleanupParams,
    db: Session = Depends(get_db)
):
    """清理日志"""
    try:
        log_service = LogService(db)
        result = log_service.cleanup_logs(cleanup_params)

        return success_response(
            data=result,
            message="日志清理完成"
        )
    except Exception as e:
        logger.error(f"清理日志失败: {e}")
        return error_response(message="清理日志失败", error_code=str(e))

@router.post("/export", response_model=dict, summary="导出日志")
async def export_logs(
    log_type: str = Query(..., description="日志类型"),
    start_time: Optional[datetime] = Query(None, description="开始时间"),
    end_time: Optional[datetime] = Query(None, description="结束时间"),
    format: str = Query("csv", description="导出格式"),
    db: Session = Depends(get_db)
):
    """导出日志"""
    try:
        log_service = LogService(db)

        # 构建过滤条件
        filters = {}
        if start_time:
            filters['start_time'] = start_time
        if end_time:
            filters['end_time'] = end_time

        # 导出日志
        export_result = log_service.export_logs(
            log_type=log_type,
            filters=filters,
            format=format
        )

        return success_response(
            data=export_result,
            message="日志导出成功"
        )
    except Exception as e:
        logger.error(f"导出日志失败: {e}")
        return error_response(message="导出日志失败", error_code=str(e))

# 实时日志
@router.get("/realtime", response_model=dict, summary="获取实时日志")
async def get_realtime_logs(
    log_type: str = Query("system", description="日志类型"),
    level: Optional[str] = Query(None, description="日志级别"),
    limit: int = Query(100, ge=1, le=500, description="限制数量"),
    db: Session = Depends(get_db)
):
    """获取实时日志"""
    try:
        log_service = LogService(db)

        # 获取最新日志
        logs = log_service.get_realtime_logs(
            log_type=log_type,
            level=level,
            limit=limit
        )

        return success_response(
            data=[log.to_dict() for log in logs],
            message="获取实时日志成功"
        )
    except Exception as e:
        logger.error(f"获取实时日志失败: {e}")
        return error_response(message="获取实时日志失败", error_code=str(e))

# 日志搜索
@router.get("/search", response_model=dict, summary="搜索日志")
async def search_logs(
    query: str = Query(..., description="搜索查询"),
    log_type: Optional[str] = Query(None, description="日志类型"),
    page: int = Query(1, ge=1, description="页码"),
    size: int = Query(50, ge=1, le=200, description="每页大小"),
    start_time: Optional[datetime] = Query(None, description="开始时间"),
    end_time: Optional[datetime] = Query(None, description="结束时间"),
    db: Session = Depends(get_db)
):
    """搜索日志"""
    try:
        log_service = LogService(db)

        # 构建过滤条件
        filters = {
            'search': query
        }
        if start_time:
            filters['start_time'] = start_time
        if end_time:
            filters['end_time'] = end_time

        # 搜索日志
        logs, total = log_service.search_logs(
            log_type=log_type,
            filters=filters,
            page=page,
            size=size
        )

        # 分页信息
        pagination = paginate(total, page, size)

        return success_response(
            data={
                "items": [log.to_dict() for log in logs],
                "pagination": pagination
            },
            message="搜索日志成功"
        )
    except Exception as e:
        logger.error(f"搜索日志失败: {e}")
        return error_response(message="搜索日志失败", error_code=str(e))
